include/dlaf/communication/error.h

```diff
@@ -0,0 +1,46 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <cstdio>
+#include <iostream>
+#include <memory>
+
+#include <mpi.h>
+
+namespace dlaf {
+namespace internal {
+
+/// MPI provides two predefined error handlers
+///
+/// - `MPI_ERRORS_ARE_FATAL` (the default ?): causes MPI to abort
+/// - `MPI_ERRORS_RETURN`: causes MPI to return an error values instead of aborting
+///
+/// This function is only relevant when the error handler is `MPI_ERRORS_RETURN`. To set the error
+/// handler use `MPI_Errhandler_set`.
+///
+inline void mpi_call(int err, char const* file, int line) {
```

Comment:
Now we have `SOURCE_LOCATION` and a trivial "pre-C++20" implementation of `source_location`

https://github.com/eth-cscs/DLA-Future/blob/095997946b7799179ec7526f1d4631811dffdd56/include/dlaf/common/utils.h#L23-L45

I would go for using it.